services/kuber/server/server.go

```diff
@@ -54,6 +57,68 @@ type server struct {
 	pb.UnimplementedKuberServiceServer
 }
 
+func (s *server) PatchClusterInfoConfigMap(_ context.Context, req *pb.PatchClusterInfoConfigMapRequest) (*pb.PatchClusterInfoConfigMapResponse, error) {
```

Comment:
hmm... I'm not really familiar with the patch command.

The config map looks like this:
```json
{
    "apiVersion": "v1",
    "data": {
        "jws-kubeconfig-6e9vpx": "..token...",
        "kubeconfig": "apiVersion: v1\nclusters:\n- cluster:\n    certificate-authority-data: ...certifiacte_authority_data...\n    server: https://0.0.0.0:6443\n  name: \"\"\ncontexts: null\ncurrent-context: \"\"\nkind: Config\npreferences: {}\nusers: null\n"
    },
    "kind": "ConfigMap",
    "metadata": {
        "creationTimestamp": "2030-09-27T09:49:01Z",
        "name": "cluster-info",
        "namespace": "kube-public",
        "resourceVersion": "360",
        "uid": "....uid...."
    }
}
```

The kubeconfig is stored in the "data" however its' not a json its yaml so to change the certificate-authority-data and server I'll need to parse it again.

basically working with 2 yamls where one is nested and only parts of the nested one needs to be changed. I tried replacing the whole kubeconfig but that didn't work not sure why.

Reply:
I am not so sure it can patch the nested value. However, I am surprised that replacing the kubeconfig did not work. Is there any other diff between them, other than EP and certs?